src/reversediff.jl

```diff
@@ -240,36 +254,60 @@ end
 # zero mean,, constant variance
 MvLogNormal(d::Int, σ::TrackedReal) = TuringMvLogNormal(TuringMvNormal(d, σ))
 
-Dirichlet(alpha::TrackedVector) = TuringDirichlet(alpha)
+# Dirichlet
+
+Dirichlet(alpha::AbstractVector{<:TrackedReal}) = TuringDirichlet(alpha)
 Dirichlet(d::Integer, alpha::TrackedReal) = TuringDirichlet(d, alpha)
 
+function _logpdf(d::Dirichlet, x::AbstractVector{<:TrackedReal})
+    return _logpdf(TuringDirichlet(d.alpha, d.alpha0, d.lmnB), x)
+end
+function logpdf(d::Dirichlet, x::AbstractMatrix{<:TrackedReal})
+    return logpdf(TuringDirichlet(d.alpha, d.alpha0, d.lmnB), x)
+end
+function loglikelihood(d::Dirichlet, x::AbstractMatrix{<:TrackedReal})
+    return loglikelihood(TuringDirichlet(d.alpha, d.alpha0, d.lmnB), x)
+end
+
+# default definition of `loglikelihood` yields gradients of zero?!
+# upstream bug caused by `view` + `track`: https://github.com/JuliaDiff/ReverseDiff.jl/pull/164
+function loglikelihood(d::TuringDirichlet, x::AbstractMatrix{<:TrackedReal})
+    return sum(i -> logpdf(d, x[:, i]), axes(x, 2))
+end
+
 for func_header in [
-    :(simplex_logpdf(alpha::TrackedVector, lmnB::Real, x::AbstractVector)),
+    :(simplex_logpdf(alpha::AbstractVector{<:TrackedReal}, lmnB::Real, x::AbstractVector)),
```

Comment:
No, it's just 1) we were not able to test these definitions before since `simplex_logpdf` was not imported in the ReverseDiff module, i.e., they define a local `simplex_logpdf` instead of `DistributionsAD.simplex_logpdf` and 2) that currently `view`s of TrackedArray create an `AbstractArray{<:TrackedReal}` but not a `TrackedArray` and hence this covers more cases (see also the ReverseDiff PR).